### PR TITLE
Allow the creator of the Course Proposal to renew it

### LINKED ISSUE #890: Fix dropdowns needing a page reload to select the same option
Using the dropdown form the "Course Management" page: If you click "Renew" from the dropdown and you close the renewal modal that pops up you are not able to select "Renew" again without manually reloading the page.

We should not require the user to reload their web page if the accidentally closed the modal. 

This was solved on Course Management page (PR #937). Make that fix work throughout the application.

### LINKED ISSUE #891: Fix Faculty SLC Proposal Renewal error
Faculty should be able to renew proposals when they have created that proposal or are listed as the "Faculty" that will be teaching the SL Course. 

If a faculty member tires to renew a SLC Proposal they are the "Faculty" of they get an error flash "Renewal Unsuccessful" and if they try and renew a proposal they created but are not listed as the "Faculty" they get an error flash "Unauthorized to preform this action".

### PR DESCRIPTION
**Pull request by stevensonmichel**
Check whether the current user has created the course and allow him to renew a Course Proposal.
We were unable to recreate the bug where faculty members could not renew proposals for which they were among the faculty. 
This feature was working properly and looks fine in the code. 

Resolves #891

### CODE REVIEW

**Review by AndersonStettner — CHANGES_REQUESTED** [review]
Good work fixing the unauthorized part of the issue. The first part, "Renewal Unsuccessful" error, is still repeatable on my end. It occurs when a faculty member hits the `Renew` button before selecting a term from the `Select Term` drop-down.

**Review by BrianRamsay — DISMISSED** [review]

**Comment by hoerstl** [comment]
We've finished fixing all of the noted issues. We are ready to be pulled.

Resolves #891

File: app/static/js/slcManagement.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-    $('.form-select').val('---');
-  });
-  $("#withdrawBtn").on("click", function() {
-    withdraw();
+  $('.modal').on('hidden.bs.modal', function () {
+    resetAllSelections()
   });
+  $('#rMTermSelect').on('change', function(){
+    if ($('#rMTermSelect').value != "---"){
+      console.log("This should enable the thing!")
```
Comment (AndersonStettner) [inline app/static/js/slcManagement.js]: Remove debug code.

File: app/templates/serviceLearning/slcManagement.html [hunk 2]
```diff
@@ -94,37 +93,31 @@ <h5 class="modal-title" id="renewModalLabel">Renew Course</h5>
...
                 <td class="fw-bold">Course Name</td>
-                <td> {{courseDict[course]['name']}}</td>
+                <td id="rMCourseNameCell"></td>
               </tr>
               <tr>
                 <td class="fw-bold"> Instructor(s)</td>
-                <td>
-                  {{courseDict[course]['faculty']|join(", ")}}
-                </td>
+                <td id="rMFacultyCell"></td>
```
Comment (AndersonStettner) [inline app/templates/serviceLearning/slcManagement.html]: Change to add a bit of specificity.

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

**Comment by hoerstl** [comment]
This branch resolves #890 and also resolves #891.

File: app/templates/serviceLearning/slcManagement.html [hunk 3]
```diff
@@ -51,6 +51,7 @@ <h1 class="text-center mb-2">{{user.firstName}} {{user.lastName}}'s SLC Proposal
       {% endfor %}
     </tbody>
   </table>
+  <input id="courseID" type="hidden" value="">
```
Comment (AndersonStettner) [inline app/templates/serviceLearning/slcManagement.html]: What is the point of the empty value?

File: app/templates/serviceLearning/slcManagement.html [hunk 4]
```diff
@@ -68,21 +69,19 @@ <h5 class="modal-title" id="withdrawModalLabel">Withdraw Course</h5>
...
-        <input id="courseID" type="hidden" value="">
         <p>Are you sure you want to withdraw this service-learning course proposal?<br><br>
           <span class="text-danger bi bi-exclamation-triangle-fill"></span>
           <b>Note: This action is irreversible. Once withdrawn, the proposal will be forever deleted.</b>
         </p>
       </div>
       <div class="modal-footer justify-content-between">
         <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Cancel</button>
-        <button type="button" class="btn btn-primary" id="withdrawBtn">Withdraw</button>
+        <button type="button" class="btn btn-primary" id="withdrawBtn" onclick="withdraw()">Withdraw</button>
```
Comment (AndersonStettner) [inline app/templates/serviceLearning/slcManagement.html]: Why did you add the onclick?

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

**Review by AndersonStettner — APPROVED** [review]

**Pull request merged**